src/Context/LivewireContext.php

```diff
@@ -2,109 +2,60 @@
 
 namespace Cockpit\Context;
 
+use Cockpit\Context\Livewire\HasLivewireManager;
+use Cockpit\Context\Livewire\HasRequestAccess;
+use Cockpit\Context\Livewire\LivewireInformationV2;
+use Cockpit\Context\Livewire\LivewireInformationV3;
 use Cockpit\Interfaces\ContextInterface;
-use Exception;
-use Illuminate\Foundation\Application;
-use Illuminate\Http\Request;
-use Illuminate\Support\Arr;
-use Livewire\LivewireManager;
+use Illuminate\Support\Facades\Log;
 
 class LivewireContext implements ContextInterface
 {
-    protected $app;
-
-    protected $livewireManager;
-
-    public function __construct(Application $app)
-    {
-        $this->app = $app;
-
-        if ($this->isRunningLivewire()) {
-            $this->livewireManager = $this->app->make(LivewireManager::class);
-        }
-    }
+    use HasRequestAccess;
+    use HasLivewireManager;
 
     public function getContext(): array
     {
-        if ($this->app->runningInConsole() && !app()->runningUnitTests() || !$this->isRunningLivewire()) {
+        if (app()->runningInConsole() && !app()->runningUnitTests() || !$this->isRunningLivewire()) {
             return [];
         }
 
-        return $this->getRequestData() + $this->getLivewireInformation();
+        $livewireInformation = match ($this->livewireVersion()) {
+            'v2'    => new LivewireInformationV2(),
+            'v3'    => new LivewireInformationV3(),
```

Comment:
We may try to use the Binding Interfaces to Implementation, using the AppServiceProvider. This would allow to have any concrete implementations, but resolving from container, so we can just typehint LivewireInformation 💡 

https://laravel.com/docs/10.x/container#binding-interfaces-to-implementations

Reply:
Can you provide an example where should I put this bind? Seems like too for nothing, but maybe I'm understanding wrong 👍

Reply:
For now we can leave it as it is, because it has a dependency on the request to get the Livewire version. We can take a look at changing this later on, because it requires a change in the architecture, using interfaces and helpers, and indeed it is a bit too much at this time.